src/helpers/script_helpers.sh

```diff
@@ -14,6 +14,7 @@ function command_steps {
   echo_if_all_true "change_directory $(git_root)" "$RUN_IN_GIT_ROOT" "$IN_SUB_FOLDER"
   echo_if_all_true "stash_open_changes" "$STASH_OPEN_CHANGES" "$INITIAL_OPEN_CHANGES"
   steps
+  echo_if_all_true "set_previous_branch \$(expected_previous_branch)"
```

Comment:
Yeah this is interesting. At the time the `$STEPS_FILE` is compiled, the output of `$(expected_previous_branch)` is meaningless. `expected_previous_branch` needs to be run after the `steps` have been run, so that we can compute the history based on what has actually happened.

By escaping with a `\`, the execution is deferred until the correct moment.